Corona-Warn-App/src/main/java/de/rki/coronawarnapp/covidcertificate/person/ui/details/PersonDetailsEvents.kt

```diff
@@ -8,4 +9,6 @@ sealed class PersonDetailsEvents
 data class OpenVaccinationCertificateDetails(val containerId: VaccinationCertificateContainerId) : PersonDetailsEvents()
 data class OpenTestCertificateDetails(val containerId: TestCertificateContainerId) : PersonDetailsEvents()
 data class OpenRecoveryCertificateDetails(val containerId: RecoveryCertificateContainerId) : PersonDetailsEvents()
+data class ValidationStart(val containerId: CertificateContainerId) : PersonDetailsEvents()
+data class ShowErrorDialog(val error: Throwable) : PersonDetailsEvents()
```

Comment:
In most cases prefer using `Exception` over `Throwable`, the difference would be that it can no longer accept `Error : Throwable`, but `Error` are usually not recoverable either and we should just crash.

Reply:
I did not  get it , I don't what value could an` Exception` bring here. also 
`Throwable.toErrorDialogBuilder(context: Context) ` is actually handling throwable

Reply:
It's not a blocker, just not good practice: Catching Throwable; as you shouldn't catch `Error` (error is `Error : Throwable`) (i.e. https://stackoverflow.com/questions/2274102/difference-between-using-throwable-and-exception-in-a-try-catch).

The error dialog builder works with throwable as there is no downside to making this possible (although it may be missleading if it encourages catching `Throwable`).

We already catch `Throwable`  in a few places and we should refactor it at some point. It's generally okay to "accept" Throwables in data classes or e.g. the `toErrorDialogBuilder` as it makes these APIs more powerful, but catching throwables, and thus catching `Error` maybes the app more vulnerable to undefined behavior if it doesn't crash in on `Error`.

Exceptions (:smile:) to that would be top-level error handling code, e.g. debug log, error logging etc. In all other cases, a Throwable that is not `Exception : Throwable` should just crash the app.

I don't think we've had this issue in practice anywhere yet, so this is all a theoretical exercise, but it would be good code quality if we do it like that :smiling_face_with_three_hearts:.

Reply:
I don't handle throwable btw, I just use this data class for showing error dialog purpose only instead of creating a specific errorLiveData for that. it is an event for the UI to the dialog ,but the actual error handling is catching exceptions, that is why I did not really think that it would make any difference in this case 

https://github.com/corona-warn-app/cwa-app-android/pull/3588/files#diff-d8aecc75f9c23ee2a9d44a769962d8c734ba5735cb76923ece40fa0992a0eae2R122-R125

Reply:
Ah then we missunderstood each other. I mean that if we reduce the options to use `Throwable` where it shouldn't be used, we discourage other developers from catching `Throwable`.